python/foglamp/services/core/api/notification.py

```diff
@@ -0,0 +1,524 @@
+            "notification_type": {
+                "description": "Type of notification",
+                "type": "enumeration",
+                "options": NOTIFICATION_TYPE,
+                "default": notification_type,
+            },
+            "enable": {
+                "description": "Enabled",
+                "type": "boolean",
+                "default": is_enabled,
+            }
+        }
+        await _update_configurations(config_mgr, notif, notification_config, rule_config, delivery_config)
+    except Exception as e:
+        raise web.HTTPBadRequest(reason=str(e))
+    else:
+        # TODO: Start notification after update
+        return web.json_response({'result': "Notification {} updated successfully".format(notif)})
+
+
+async def delete_notification(request):
+    """ Delete an existing notification
+
+    :Example:
+        curl -X DELETE http://localhost:8081/foglamp/notification/<notification_name>
+    """
+    try:
+        notification_service = ServiceRegistry.get(s_type=ServiceRecord.Type.Notification.name)
+        _address, _port = notification_service[0]._address, notification_service[0]._management_port
+    except service_registry_exceptions.DoesNotExist:
+        raise web.HTTPNotFound(reason="No Notification service available.")
+
+    try:
+        notif = request.match_info.get('notification_name', None)
+        if notif is None:
+            raise web.HTTPInternalServerError(reason="Notification name is required for deletion.")
+
+        url = str(request.url)
+        url_parts = url.split("/foglamp/notification")
+        url = '{}/foglamp/notification/{}'.format(url_parts[0], urllib.parse.quote(notif))
+        notification = json.loads(await _hit_get_url(url))
+
+        # TODO: Stop notification before deletion
+
+        # Removes the child categories for the rule and delivery plugins, Removes the category for the notification itself
+        storage = connect.get_storage_async()
+        config_mgr = ConfigurationManager(storage)
+        await _delete_configuration(storage, config_mgr, notif)
+
+        audit = AuditLogger(storage)
+        await audit.information('NTFDL', notification)
+    except Exception as ex:
+        raise web.HTTPInternalServerError(reason=str(ex))
+    else:
+        return web.json_response({'result': 'notification {} deleted successfully.'.format(notif)})
+
+
+async def _hit_get_url(get_url):
+    try:
+        async with aiohttp.ClientSession() as session:
+            async with session.get(get_url) as resp:
+                status_code = resp.status
+                jdoc = await resp.text()
+                if status_code not in range(200, 209):
+                    _logger.error("Error code: %d, reason: %s, details: %s, url: %s", resp.status, resp.reason, jdoc, get_url)
+                    raise StorageServerError(code=resp.status, reason=resp.reason, error=jdoc)
+    except Exception as ex:
+        raise web.HTTPInternalServerError(reason=ex)
```

Comment:
I think this is not appropriate to through, we should log and raise only whatever exception occured

Reply:
The reason is that this hit failed and therefore we must not proceed further. That is why it is logical to raise an error.